### ISSUE TITLE
sf::Music::openFromStream and sf::SoundBuffer::loadFromStream fails for non-flac files

### ISSUE THREAD

**Issue by MorleyDev** [LaurentGomila/SFML#822]
Loading Music or SoundBuffer through an InputStream fails, but loading straight from the file with openFromFile/loadFromFile succeeds.

Code which recreates (orchestral.ogg from examples)
```cpp
#include <SFML/Audio.hpp>
#include <SFML/System/FileInputStream.hpp>
#include <iostream>

int main() {
    std::cout << "Starting..." << std::endl;
    sf::Music fileMusic;
    if (!fileMusic.openFromFile("orchestral.ogg")) {
        std::cerr << "Too early boom?"  << std::endl;
        return 3;
    }

    sf::FileInputStream stream;
    sf::Music music;
    if (!stream.open("orchestral.ogg")) {
        std::cerr << "Too early boom?"  << std::endl;
        return 2;
    }
    if (!music.openFromStream(stream)) {
        std::cerr << "BOOM!"  << std::endl;
        return 1;
    }
    std::cout << "Done" << std::endl;
    return 0;
}
```

**Comment by MarioLiebisch** [comment]
Is there any reason you "reimplemented" `sf::FileStream`? Did you ensure the problem isn't caused by the copied code?

**Comment by LaurentGomila** [comment]
The discussion is still pending on the forum, this issue will be reopened when things are confirmed.

By the way, ```SoundEffect``` doesn't exist, did you mean ```SoundBuffer```?

**Comment by eXpl0it3r** [comment]
It's indeed an issue with SFML. The [SoundFileFactory.cpp](https://github.com/LaurentGomila/SFML/blob/master/src/SFML/Audio/SoundFileFactory.cpp#L120) is missing `stream.seek(0);`. Also the comment should get updated, since when working with a stream there are not "filenames". :wink:

The above example should read as follows, since `sf::Music` uses the stream.

```cpp
#include <SFML/Audio.hpp>
#include <SFML/System/FileInputStream.hpp>
#include <iostream>
#include <cstdio>

int main() {
    std::cout << "Starting..." << std::endl;
    sf::Music fileMusic;
    if (!fileMusic.openFromFile("orchestral.ogg")) {
        std::cerr << "Too early boom?"  << std::endl;
        return 3;
    }

    sf::FileInputStream stream;
    sf::Music music;
    if (!stream.open("orchestral.ogg")) {
        std::cerr << "Too early boom?"  << std::endl;
        return 2;
    }
    if (!music.openFromStream(stream)) {
        std::cerr << "BOOM!"  << std::endl;
        return 1;
    }
    std::cout << "Done" << std::endl;
}
```

**Comment by eXpl0it3r** [comment]
Superseded by #823
